core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -829,17 +822,15 @@ private ExprValue exprYear(ExprValue date) {
   }
 
   /**
-   * Prepare LocalDateTime value.
-   * @param supplier A function which returns LocalDateTime to format.
+   * Prepare LocalDateTime value. Truncate fractional second part according to the argument.
    * @param fsp argument is given to specify a fractional seconds precision from 0 to 6,
    *            the return value includes a fractional seconds part of that many digits.
    * @return LocalDateTime object.
    */
-  private LocalDateTime formatLocalDateTime(Supplier<LocalDateTime> supplier,
-                                            @Nullable Integer fsp) {
-    var res = supplier.get();
+  private LocalDateTime formatNow(@Nullable Integer fsp) {
+    var res = LocalDateTime.now();
     if (fsp == null) {
-      return res;
+      fsp = 0;
     }
     var defaultPrecision = 9; // There are 10^9 nanoseconds in one second
     if (fsp < 0 || fsp > 6) { // Check that the argument is in the allowed range [0, 6]
```

Comment:
Do you need this if you are not having the values in now be handled?

Reply:
`sysdate()` uses the same function under the hood and it has an argument.
Try
```sql
select sysdate(), sysdate(0), sysdate(6);
```